### PR TITLE
IGNITE-17541 Create get and set methods with serverExceptionStackTrac…

### PR DESCRIPTION
**Pull request by J-Bakuli**
[IGNITE-13389](https://issues.apache.org/jira/browse/IGNITE-13389) introduces ThinClientConfiguration#sendServerExceptionStackTraceToClient(), but name of the method does not allow to set up this option in XML configuration, because Spring expects "set" prefix for setters.
https://github.com/apache/ignite/blob/4a6ff6f4aa1994b39bf40cc8ad80ac5df704d569/modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java#L107

Possible workarounds:

1. Add extra ThinClientConfiguration bean with necessary parameters

2. Set up thinClientConfiguration by means of SpEL

Although there several ways out of the situation it is more convenient to set up this property directly and use 'set' and 'get' prefixes to setter and getter methods thus I have added 'set' prefix to the sendServerExceptionStackTraceToClient() below:

1. ThinClientConfiguration.class
https://github.com/apache/ignite/blob/4a6ff6f4aa1994b39bf40cc8ad80ac5df704d569/modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java#L107

2. PlatformConfigurationUtils.class in readClientConnectorConfiguration()
https://github.com/apache/ignite/blob/4a6ff6f4aa1994b39bf40cc8ad80ac5df704d569/modules/core/src/main/java/org/apache/ignite/internal/processors/platform/utils/PlatformConfigurationUtils.java#L1963

Although there are some workarounds it would be more convenient to set up this property directly.

We should add "set" and "get" prefix to setter and getter methods respectively or add extra methods with such prefixes from the point of possible compatibility issues (see [IGNITE-16549](https://issues.apache.org/jira/browse/IGNITE-16549)).

Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

**Comment by J-Bakuli** [comment]
@alex-plekhanov, could you please review my changes?

File: modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java [hunk 1]
```diff
@@ -103,22 +112,45 @@ public ThinClientConfiguration setMaxActiveComputeTasksPerConnection(int maxActi
...
+     */
+    public boolean getServerExceptionStackTraceToClient() {
+        return serverExcStackTraceToClient;
+    }
+
+    /**
+     * @param serverExcStackTraceToClient if {@code true} sets a server exception stack that is to be sent to the client side.
+     * @return {@code this} for chaining.
+     */
+    public ThinClientConfiguration setServerExceptionStackTraceToClient(boolean serverExcStackTraceToClient) {
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java:145]: Is not clear from the method's name what exactly this method set (javadoc also confusing). How about `setServerToClientExceptionStackTraceSending(boolean sendStackTrace)`?

File: modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java [hunk 2]
```diff
@@ -103,22 +112,45 @@ public ThinClientConfiguration setMaxActiveComputeTasksPerConnection(int maxActi
...
+        this.serverExcStackTraceToClient = serverExcStackTraceToClient;
+
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java:152]: Redundant change, let's keep one-line javadoc.

File: modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java (hunk 1 above)
Comment (J-Bakuli) [reply modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java:145]: Agree, I have changed the method name as per your recommendation above.
Java doc then is "@param sendStackTrace if {@code true} sets a server to client exception stack trace sending." 
Do you agree?

File: modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java [hunk 3]
```diff
@@ -48,10 +56,11 @@ public ThinClientConfiguration() {
     }
 
     /**
-     * Creates thin-client configuration by copying all properties from given configuration.
-     *
      * @param cfg Configuration to copy.
+     * Creates thin-client configuration by copying all properties from given configuration.
+     * @deprecated
      */
+    @Deprecated
```
[the remark this answers is not in the document]
Comment (J-Bakuli) [reply modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java:63]: It used the deprecated field sendServerExcStackTraceToClient. 
After your comments I have removed the Deprecated mark so this constructor is no longer deprecated.

File: modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java [hunk 4]
```diff
@@ -37,9 +37,17 @@ public class ThinClientConfiguration {
...
+    /**
+     * @see new field serverExcStackTraceToClient.
+     * If {@code true} sends a server exception stack trace to the client side.
+     * @deprecated please now use new field serverExcStackTraceToClientbecause.
+     */
+    @Deprecated
     private boolean sendServerExcStackTraceToClient;
 
+    /** If {@code true} a server exception stack trace is sent to the client side. */
+    private boolean serverExcStackTraceToClient;
```
[the remark this answers is not in the document]
Comment (J-Bakuli) [reply modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java:49]: Thank you, I have changed the code to be in line with your comments above.
That new field was added to be inserted into another constructor that is to be used later when new pull requests remove deprecated fields / methods from the code completely. That was my intention.

This PR and the ticket is to add set/get methods w/o overcomplication so agree with you.

File: modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java (hunk 1 above)
Comment (alex-plekhanov) [reply modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java:145]: Javadoc for this parameter of the old method looks better to me, but it's up to you. 
Also, the first word of the javadoc should start with the upper case letter.

File: modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java [hunk 5]
```diff
@@ -103,21 +103,42 @@ public ThinClientConfiguration setMaxActiveComputeTasksPerConnection(int maxActi
...
     public ThinClientConfiguration sendServerExceptionStackTraceToClient(boolean sendServerExcStackTraceToClient) {
         this.sendServerExcStackTraceToClient = sendServerExcStackTraceToClient;
 
         return this;
     }
 
+    /**
+     * @return If {@code true} gets a server exception stack that is to be sent to the client side.
+     */
+    public boolean getServerExceptionStackTraceToClient() {
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java:128]: The name should be the same as for setter. Javadoc also must be fixed. 
BTW do we really need getter with `get` prefix? We need setter to set value in spring XML configuration, but getter is not used here, pehaps the new getter is redundant?

Comment (J-Bakuli) [reply modules/core/src/main/java/org/apache/ignite/configuration/ThinClientConfiguration.java:128]: Thank you, I will rename the method to be the same as for setter and correct Javadoc

**Comment by J-Bakuli** [comment]
@NSAmelchev, could you please review the changes? 
They are almost fully checked by the previous reviewer.
Green visa https://mtcga.gridgain.com/prs.html

**Review by alex-plekhanov — APPROVED** [review]

**Pull request closed without merging**